api/src/main/kotlin/com/google/devtools/ksp/utils.kt

```diff
@@ -64,7 +64,8 @@ fun KSClassDeclaration.getDeclaredProperties(): List<KSPropertyDeclaration> {
 }
 
 fun KSClassDeclaration.getConstructors(): List<KSFunctionDeclaration> {
-    return this.getDeclaredFunctions().filter { it.simpleName.asString() == this.simpleName.asString() }
+    val functions = if (this.origin == Origin.JAVA) this.getAllFunctions() else this.getDeclaredFunctions()
```

Comment:
`getAllFunctions` sounds like an overkill.

Reply:
This will only be for Java classes, and it can't be avoided as for Java classes, default constructor has to be looked up in descriptor. Alternatively we can add a `getConstructors` to `KSClassDeclaration` to save additional memory usage from `getAllFunctions()`, but the biggest overhead in resolving can't be avoided.